.github/workflows/ci-tests.yml

```diff
@@ -0,0 +1,50 @@
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Tests
+
+on:
+  push:
+    branches: [ master, refactor-api ]
+  pull_request:
+    branches: [ master, refactor-api ]
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        platform: [ubuntu-latest, macos-latest, windows-latest]
+        python-version: [3.7]
+        include:
+          - platform: ubuntu-latest
+            python-version: 3.8
+    runs-on: ${{ matrix.platform }}
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install package with dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -e .[tests]
+# Uncomment if sphinx docs are available in the repository
+#        pip install -e .[tests,docs]
+    - name: Lint with flake8
+      run: |
+        # exit-zero treats all errors as warnings.
+        flake8 . --count --exit-zero --statistics --show-source
```

Comment:
Does ``-exits-zero` mean that the `flake8` run never fails? If yes, how does the warning look like in a PR? Do I see it?

Reply:
I made a change now, where we fail the build for "major" issues, but only print warnings to the log for style related issues. 
The log can be found at the action view. 
see the comment